### DISCUSSION TITLE
Is there official (or any) list of supported browsers by lit-html 2.0 somewhere?

### DISCUSSION THREAD

**Discussion by anwerso** [lit/lit#2320] [Q&A] [answered]
I'm trying to migrate from v1.0 and found that lit-html is not working on Chrome 49 (yes, we have decent amount of users on Windows XP). I polyfilled `globalThis` and `Element.prototype.getAttributeNames` and all seems to start working. Wonder what other problems can arise later?

p.s.
I only need standalone version support, not using whole lib with LitElement.

**Comment by justinfagnani** [comment, marked as answer]
We list what we test on. Lit may work on older browsers, but without continuous testing on them, we can't make any guarantees.

If thinks work now, that's a good sign, because we're pretty stable in the core libraries - and we used to test on Safari 9 and Chrome 41 because they approximated older web crawlers - but it's entirely possible that we start using a new-ish API that's available in our supported browser list, but not in Chrome 49 and you'd have to add another polyfill.

All I can recommend is that you do continuous testing of new versions of Lit as they come out. We will try to support older browsers on a as-is, very low priority basis - that is, we'll consider PRs that fix issues in older browsers, but only if they have little-to-no cost to modern browsers in terms of perf and code size. Hope that helps!

**Reply by anwerso** [reply]
`self` is a good alternative of `globalThis` if we are talking about browser environment. On second thought there is SSR somewhere... well, it's one line check/polyfill anyway. 

`getAttributeNames` can be easily polyfilled, and probably it's a good idea to add section somewhere in "Requirements" documentation. Or downgrade to a `Element.attributes` in code when unavailable. I'll think about PR if no more problems appears.

**Reply by justinfagnani** [reply]
`getAttributeNames` is used specifically to avoid materializing Attr nodes when all we want is the names. It's a performance concern, and code size concern to include the fallback code in core. Requiring a polyfill is what we want to do there.
